tests/scripts/run-throughput-perf.py

```diff
@@ -0,0 +1,361 @@
+    build_type = next((b for b in valid_build_types if b.lower() == build_type.lower()), build_type)
+
+    validate_arg(arch, lambda item: item in valid_archs)
+    validate_arg(build_type, lambda item: item in valid_build_types)
+    validate_arg(run_type, lambda item: item in valid_run_types)
+    validate_arg(operating_system, lambda item: item in valid_os)
+
+    if clr_root is None:
+        clr_root = nth_dirname(os.path.abspath(sys.argv[0]), 3)
+    else:
+        clr_root = os.path.normpath(clr_root)
+        validate_arg(clr_root, lambda item: os.path.isdir(clr_root))
+
+    if assembly_root is None:
+        # If unspecified, assembly_root will be assumed to be the Core_Root directory
+        assembly_root = os.path.join(clr_root, 'bin', 'tests', operating_system + '.' + arch + '.' + build_type, 'Core_Root')
+    else:
+        assembly_root = os.path.normpath(assembly_root)
+        validate_arg(assembly_root, lambda item: os.path.isdir(assembly_root))
+
+    if not benchview_path is None:
+        benchview_path = os.path.normpath(benchview_path)
+        validate_arg(benchview_path, lambda item: os.path.isdir(benchview_path))
+
+    args = (arch, operating_system, build_type, run_type, clr_root, assembly_root, benchview_path)
+
+    # Log configuration
+    log('Configuration:')
+    log(' arch: %s' % arch)
+    log(' os: %s' % operating_system)
+    log(' build_type: %s' % build_type)
+    log(' run_type: %s' % run_type)
+    log(' clr_root: %s' % clr_root)
+    log(' assembly_root: %s' % assembly_root)
+    if not benchview_path is None:
+        log('benchview_path : %s' % benchview_path)
+
+    return args
+
+def nth_dirname(path, n):
+    """ Find the Nth parent directory of the given path
+    Args:
+        path (str): path name containing at least N components
+        n (int): num of basenames to remove
+    Returns:
+        outpath (str): path with the last n components removed
+    Notes:
+        If n is 0, path is returned unmodified
+    """
+
+    assert n >= 0
+
+    for i in range(0, n):
+        path = os.path.dirname(path)
+
+    return path
+
+def del_rw(action, name, exc):
+    os.chmod(name, stat.S_IWRITE)
+    os.remove(name)
+
+def log(message):
+    """ Print logging information
+    Args:
+        message (str): message to be printed
+    """
+
+    print('[%s]: %s' % (sys.argv[0], message))
+
+def generateCSV(dll_name, dll_runtimes):
+    """ Write throuput performance data to a csv file to be consumed by measurement.py
+    Args:
+        dll_name (str): the name of the dll
+        dll_runtimes (float[]): A list of runtimes for each iteration of the performance test
+    """
+
+    csv_file_name = "throughput_%s.csv" % (dll_name)
+    csv_file_path = os.path.join(os.getcwd(), csv_file_name)
+    output_file = open(csv_file_path, "w")
+
+    for iteration in dll_runtimes:
+        output_file.write("%s,%s,%s,%s\n" % ("default", "coreclr-crossgen-tp", dll_name, iteration))
+
+    output_file.close()
```

Comment:
As written, this works with both python 2 and 3, except for the calls to measurement, submission, and upload.py, which have a python 3 dependency.